specification/attestation.adoc

```diff
@@ -1,309 +1,880 @@
+
+The TVM challenge claim is a `sbi_tee_guest_get_evidence()` caller provided
+value. The semantics of this Claim is TVM implementation specific, but it is
+generally used for demonstrating Evidence freshness to a Relying Party.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TVM Challenge Claim"]
+[source,cddl]
+----
+tvm-challenge-label = 10 ; EAT nonce
+tvm-challenge-type = bytes .size 64
+
+tvm-challenge = (
+   tvm-challenge-label => tvm-challenge-type
+)
+----
+
+====== TVM Identity Claim
+
+The TVM identity claim value is a `sbi_tee_host_create_tvm()` provided argument
+
+It is used by the host TVM creator (e.g. the host VMM) to bind a TVM to an
+identity or more generically a specific piece of data (e.g. an Attestation
+Service public key, a configuration blob, etc) through its hash value.
+
+TVM identity allows for untrusted hosts to provide a TVM with unmeasured but
+attestable pieces of data. A Relying Party can then verify the TVM measurements
+separately from the host provided TVM identity.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TVM Identity Claim"]
+[source,cddl]
+----
+tvm-identity-label = TBD
+tvm-identity-type = bytes .size 64
+
+tvm-identity = (
+   tvm-identity-label => tvm-identity-type
+)
+----
+
+====== TVM Public Key Claim
+
+The TVM public key claim value is a a `sbi_tee_guest_get_evidence()` caller
+provided value. In other words, the TVM guest provides its own, generally
+ephemeral public key to the TSM to be included into the Evidence. A Relying
+Party will use that public key to encrypt secrets that are released to the
+trusted TVM.
+
+The value of the TVM public key claim is a COSE_Key <<COSE>> structure encoded
+as a CBOR bytes string.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TVM Public Key Claim"]
+[source,cddl]
+----
+tvm-public-key-label = TBD
+
+tvm-public-key = (
+   tvm-public-key-label => riscv-ap-tee-public-key-type
+)
+----
+
+====== TVM Static Measurements Claim
+
+The TVM static measurements claim value is the list of all static measurements
+for the TVM. The list must contain at most 8 entries.
+
+Each measurement in the list is a map of the following values:
+
+* *Measurement register index*: This describes the measurement register index
+used by the TSM to store the measurement value. This can be mapped to well known
+measurement register indexes like e.g. the TCG<<TCG_Client>> defined ones.
+* *Measurement value*: This is the measurement value.
+* *Hash algorithm identifier*: This describes which algorithm was used to
+generate the component measurement. It is recommended to use one of the IANA
+registered hash algorithm name<<Hash_Algorithm_Names>>.
+
+[caption="Figure {counter:image}: ", reftext="Figure {image}"]
+[title= "TVM Static Measurements Claim"]
+[source,cddl]
+----
+tvm-static-measurements-label = TBD
+riscv-ap-tee-measurement = {
+ ? 1 => uint                   ; Measurement register index
```

Comment:
is the index optional? I did not think so - then remove the "?"

Reply:
That's right, there will always be an index.